File: Node/TypeORM_ORM/src/dal/entities/TestData.ts
import { Column, Entity, Index, PrimaryGeneratedColumn } from "typeorm";

@Index("PK__TestData__8CC3310000C56314", ["id"], { unique: true })
@Entity("TestData", { schema: "dbo" })
export class TestData {
  @PrimaryGeneratedColumn({ type: "int", name: "Id" })
  id: number;

  @Column("varchar", { name: "TestName", nullable: true, length: 50 })
  testName: string | null;

  @Column("varchar", { name: "CpuUsage", nullable: true, length: 50 })
  cpuUsage: string | null;

  @Column("varchar", { name: "MemoryUsage", nullable: true, length: 50 })
  memoryUsage: string | null;

  @Column("varchar", { name: "Performance", nullable: true, length: 50 })
  performance: string | null;

  @Column("varchar", { name: "Language", nullable: true, length: 50 })
  language: string | null;
}

File: Go/GORM_ORM/dal/entities/testData.go
package entities

type TestData struct {
	ID          int     `gorm:"primaryKey;autoIncrement;column:Id"`
	TestName    *string `gorm:"size:50;column:TestName"`
	CpuUsage    *string `gorm:"size:50;column:CpuUsage"`
	MemoryUsage *string `gorm:"size:50;column:MemoryUsage"`
	Performance *string `gorm:"size:50;column:Performance"`
	Language    *string `gorm:"size:50;column:Language"`
}

func (TestData) TableName() string {
	return "TestData"
}
